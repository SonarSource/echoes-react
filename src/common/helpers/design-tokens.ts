/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import designTokensColors from '~generated/design-tokens-colors.json';

export type DesignTokens = keyof typeof designTokensColors;

export type DesignTokensColors = {
  [K in DesignTokens]: K extends `echoes-color-${string}` ? K : never;
}[DesignTokens];
