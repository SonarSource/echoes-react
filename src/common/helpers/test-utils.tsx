/*
 * Echoes React
 * Copyright (C) 2023-2024 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { RenderOptions, render as rtlRender } from '@testing-library/react';
import userEvent, { Options as UserEventsOptions } from '@testing-library/user-event';
import React, { ComponentProps, PropsWithChildren } from 'react';
import { IntlProvider } from 'react-intl';
import { PropsWithLabels } from '~types/utils';
import { TooltipProvider } from '../../components/tooltip';

export function render(
  ui: React.ReactElement,
  options?: RenderOptions,
  userEventOptions?: UserEventsOptions,
) {
  return {
    ...rtlRender(ui, { wrapper: ContextWrapper, ...options }),
    user: userEvent.setup(userEventOptions),
  };
}

function ContextWrapper({ children }: PropsWithChildren<{}>) {
  return (
    <IntlProvider defaultLocale="en-us" locale="en-us">
      <TooltipProvider delayDuration={0}>{children}</TooltipProvider>
    </IntlProvider>
  );
}

export type OmitPropsWithLabels<T extends React.JSXElementConstructor<any>> = Partial<
  Omit<ComponentProps<T>, keyof PropsWithLabels<{}>>
> &
  PropsWithLabels<{}>;
