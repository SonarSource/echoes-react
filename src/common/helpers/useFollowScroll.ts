/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { useEffect, useState } from 'react';
import { throttle, THROTTLE_SHORT_DELAY } from './utils';

/**
 * A React hook that tracks the scroll position of the document.
 * It returns an object containing the current scroll position (left and top).
 * The scroll position is updated at a throttled rate to improve performance.
 *
 * @returns An object with `left` and `top` properties representing the current scroll position.
 */
export function useFollowScroll(enabled = true) {
  const [left, setLeft] = useState(0);
  const [top, setTop] = useState(0);

  useEffect(() => {
    const followScroll = throttle(() => {
      if (document.documentElement) {
        setLeft(document.documentElement.scrollLeft);
        setTop(document.documentElement.scrollTop);
      }
    }, THROTTLE_SHORT_DELAY);

    if (enabled) {
      document.addEventListener('scroll', followScroll);
    }

    return () => document.removeEventListener('scroll', followScroll);
  }, [enabled]);

  return { left, top };
}
