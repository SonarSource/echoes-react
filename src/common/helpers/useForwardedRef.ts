/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { type ForwardedRef, useState } from 'react';

/**
 * This hook may be used to intercept a forwarded ref, providing a local ref
 * that can be used in the component.
 *
 * **Example**
 *
 * ```typescript
 * const Input = forwardRef<HTMLInputElement>((props, forwardedRef) => {
 *   const [ref, setRef] = useForwardedRef(forwardedRef);
 *   return <input ref={setRef} />
 * });
 * ```
 */
export function useForwardedRef<T>(forwardedRef: ForwardedRef<T>) {
  const [ref, setRef] = useState<T | null>(null);

  const setForwardedRef = (element: T | null) => {
    if (typeof forwardedRef === 'function') {
      forwardedRef(element);
    } else if (forwardedRef) {
      forwardedRef.current = element;
    }

    setRef(element);
  };

  return [ref, setForwardedRef] as const;
}
