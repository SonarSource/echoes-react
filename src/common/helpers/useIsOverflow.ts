/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { DependencyList, MutableRefObject, useLayoutEffect, useState } from 'react';

/**
 * This hook checks if the content of a given ref is overflowing its container horizontally.
 * It returns a boolean indicating whether the content is overflowing.
 *
 * @param forwardedRef - The ref to check for overflow.
 * @param deps - Optional dependencies array to re-compute the overflow state.
 */
export function useIsOverflow(
  forwardedRef: MutableRefObject<HTMLElement | null>,
  deps: DependencyList = [],
) {
  const [isOverflow, setIsOverflow] = useState<boolean | undefined>(undefined);

  useLayoutEffect(() => {
    if (forwardedRef.current) {
      setIsOverflow(forwardedRef.current.scrollWidth > forwardedRef.current.offsetWidth);
    }
    // We want to be able to pass an array of dependencies to re-compute the overflow state
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [forwardedRef, ...deps]);

  return [isOverflow] as const;
}
