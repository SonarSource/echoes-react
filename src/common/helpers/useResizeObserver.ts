/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { RefObject, useCallback, useEffect, useState } from 'react';

export function useResizeObserver(ref: RefObject<HTMLElement>) {
  const [width, setWidth] = useState<number>();
  const [height, setHeight] = useState<number>();

  const handleResize = useCallback<ResizeObserverCallback>((entries) => {
    if (!Array.isArray(entries)) {
      return;
    }

    const entry = entries[0];
    setWidth(entry.contentRect.width);
    setHeight(entry.contentRect.height);
  }, []);

  useEffect(() => {
    if (!ref.current) {
      return undefined;
    }

    const RO = new ResizeObserver(handleResize);
    RO.observe(ref.current);

    return () => {
      RO.disconnect();
    };
  }, [handleResize, ref]);

  return { width, height };
}
