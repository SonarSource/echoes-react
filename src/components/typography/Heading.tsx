/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import styled from '@emotion/styled';
import { forwardRef, PropsWithChildren } from 'react';

export enum HeadingSize {
  ExtraSmall = 'xsmall',
  Small = 'small',
  Medium = 'medium',
  Large = 'large',
  ExtraLarge = 'xlarge',
}

type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5';

export interface HeadingProps {
  as: HeadingTag;
  className?: string;
  id?: string;
  hasMarginBottom?: boolean;
  size?: `${HeadingSize}`;
}

export const Heading = forwardRef<HTMLDivElement, Readonly<PropsWithChildren<HeadingProps>>>(
  ({ as, size = defaultSizeByTag[as], hasMarginBottom = false, ...props }, ref) => (
    <StyledHeading as={as} hasMarginBottom={hasMarginBottom} ref={ref} size={size} {...props} />
  ),
);

Heading.displayName = 'Heading';

const defaultSizeByTag: Record<HeadingTag, HeadingSize> = {
  h1: HeadingSize.ExtraLarge,
  h2: HeadingSize.Large,
  h3: HeadingSize.Medium,
  h4: HeadingSize.Small,
  h5: HeadingSize.ExtraSmall,
};

const StyledHeading = styled.div<Required<Pick<HeadingProps, 'hasMarginBottom' | 'size'>>>`
  font: ${getHeadingFont};
  color: var(--echoes-color-text-strong);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);

  ${({ hasMarginBottom, size }) =>
    hasMarginBottom ? `margin-bottom: ${bottomMarginByHeadingSize({ size })}` : ''}
`;
StyledHeading.displayName = 'StyledHeading';

function getHeadingFont({ size }: Required<Pick<HeadingProps, 'size'>>) {
  return HEADING_TYPOGRAPHY_MAP[size];
}

const HEADING_TYPOGRAPHY_MAP = {
  [HeadingSize.ExtraSmall]: 'var(--echoes-typography-heading-xsmall)',
  [HeadingSize.Small]: 'var(--echoes-typography-heading-small)',
  [HeadingSize.Medium]: 'var(--echoes-typography-heading-medium)',
  [HeadingSize.Large]: 'var(--echoes-typography-heading-large)',
  [HeadingSize.ExtraLarge]: 'var(--echoes-typography-heading-xlarge)',
};

const bottomMarginByHeadingSize = ({ size }: Required<Pick<HeadingProps, 'size'>>) => {
  switch (size) {
    case HeadingSize.Large:
      return 'var(--echoes-dimension-space-200)';
    case HeadingSize.ExtraLarge:
      return 'var(--echoes-dimension-space-250)';
    default:
      return 'var(--echoes-dimension-space-100)';
  }
};
