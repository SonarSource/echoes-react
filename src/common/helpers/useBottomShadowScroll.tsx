/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import styled from '@emotion/styled';
import { RefObject, useCallback, useEffect, useState } from 'react';
import { cssVar } from '~utils/design-tokens';
import { useResizeObserver } from './useResizeObserver';

/**
 * Custom hook to manage the visibility of a bottom shadow on a scrollable container.
 * The shadow indicates that there is more content to scroll to.
 * @param scrollableContainerRef - Ref to the scrollable container element, its height and scroll event are monitored to recompute the shadow
 * @param resizableContentRef - Ref to the resizable content element inside the scrollable container, its height is monitored to recompute the shadow
 * @returns {boolean} Whether the bottom shadow should be shown
 */
export function useBottomShadowScroll(
  scrollableContainerRef: RefObject<HTMLElement>,
  resizableContentRef: RefObject<HTMLElement>,
) {
  const [showBottomShadow, setShowBottomShadow] = useState(false);

  // Monitor the height of the scrollable container, and recompute the shadow if it changes
  const { height: scrollableContainerHeight } = useResizeObserver(scrollableContainerRef);
  // Monitor the height of the resizable content, and recompute the shadow if it changes, because it may move the scroll position
  const { height: resizableContentHeight } = useResizeObserver(resizableContentRef);

  const updateBottomShadowScroll = useCallback(() => {
    if (scrollableContainerRef.current) {
      const { scrollHeight, clientHeight, scrollTop } = scrollableContainerRef.current;
      setShowBottomShadow(scrollTop + clientHeight < scrollHeight - 8); // -8px for bottom padding
    }
  }, [scrollableContainerRef]);

  // Compute the bottom shadow visibility on first render and attach the scroll event listener
  useEffect(() => {
    const scrollableContainerRefNode = scrollableContainerRef.current;
    if (!scrollableContainerRefNode) {
      return undefined;
    }

    // Re-compute the bottom shadow visibility on scroll event
    scrollableContainerRefNode.addEventListener('scroll', updateBottomShadowScroll);

    // Initial computation
    updateBottomShadowScroll();

    return () => {
      scrollableContainerRefNode.removeEventListener('scroll', updateBottomShadowScroll);
    };
  }, [scrollableContainerRef, updateBottomShadowScroll]);

  // Re-compute the bottom shadow visibility on container and content resize
  useEffect(updateBottomShadowScroll, [
    resizableContentHeight,
    scrollableContainerHeight,
    updateBottomShadowScroll,
  ]);

  return [showBottomShadow] as const;
}

export const BottomShadowScroll = styled.div`
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;

  min-height: ${cssVar('dimension-height-400')};

  border-bottom: ${cssVar('border-width-default')} solid ${cssVar('color-border-weak')};
  background: radial-gradient(
      farthest-side at 50% 100%,
      ${cssVar('bottom-scroll-colors-shadow-gradient')},
      ${cssVar('color-support-transparent')}
    )
    center bottom;
`;
BottomShadowScroll.displayName = 'BottomShadowScroll';
