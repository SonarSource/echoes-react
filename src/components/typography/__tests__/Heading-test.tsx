/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { screen } from '@testing-library/react';
import { ComponentProps } from 'react';
import { render } from '~common/helpers/test-utils';
import { Heading, HeadingSize } from '../Heading';

it('uses the default size', () => {
  const { container } = setupHeading();

  expect(container).toMatchSnapshot();
});

it.each(['h1', 'h2', 'h3', 'h4', 'h5'])(
  'renders a Heading as an %s with no bottom margin',
  async (as: ComponentProps<typeof Heading>['as']) => {
    const children = `This is a heading rendered as "${as}" with no bottom margin`;

    const { container } = setupHeading({ as, children });

    expect(await screen.findByText(children)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  },
);

it.each([
  HeadingSize.ExtraSmall,
  HeadingSize.Small,
  HeadingSize.Medium,
  HeadingSize.Large,
  HeadingSize.ExtraLarge,
])('renders a Heading with size %s and a bottom margin', async (size) => {
  const children = `This is a heading with size "${size}" with a bottom margin`;

  const { container } = setupHeading({ children, hasMarginBottom: true, size });

  expect(await screen.findByText(children)).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});

function setupHeading(
  { as, children, ...otherProps }: Partial<ComponentProps<typeof Heading>> = {
    as: 'h1',
    children: 'Text goes here',
  },
) {
  return render(
    <Heading as={as as ComponentProps<typeof Heading>['as']} {...otherProps}>
      {children}
    </Heading>,
  );
}
