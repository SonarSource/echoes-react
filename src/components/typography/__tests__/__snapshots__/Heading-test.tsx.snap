// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Heading as an h1 with no bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-xlarge);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
}

<div>
  <h1
    class="emotion-0 emotion-1"
  >
    This is a heading rendered as "h1" with no bottom margin
  </h1>
</div>
`;

exports[`renders a Heading as an h2 with no bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-large);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
}

<div>
  <h2
    class="emotion-0 emotion-1"
  >
    This is a heading rendered as "h2" with no bottom margin
  </h2>
</div>
`;

exports[`renders a Heading as an h3 with no bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-medium);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
}

<div>
  <h3
    class="emotion-0 emotion-1"
  >
    This is a heading rendered as "h3" with no bottom margin
  </h3>
</div>
`;

exports[`renders a Heading as an h4 with no bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-small);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
}

<div>
  <h4
    class="emotion-0 emotion-1"
  >
    This is a heading rendered as "h4" with no bottom margin
  </h4>
</div>
`;

exports[`renders a Heading as an h5 with no bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-xsmall);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
}

<div>
  <h5
    class="emotion-0 emotion-1"
  >
    This is a heading rendered as "h5" with no bottom margin
  </h5>
</div>
`;

exports[`renders a Heading with size large and a bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-large);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
  margin-bottom: var(--echoes-dimension-space-200);
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    This is a heading with size "large" with a bottom margin
  </div>
</div>
`;

exports[`renders a Heading with size medium and a bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-medium);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
  margin-bottom: var(--echoes-dimension-space-100);
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    This is a heading with size "medium" with a bottom margin
  </div>
</div>
`;

exports[`renders a Heading with size small and a bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-small);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
  margin-bottom: var(--echoes-dimension-space-100);
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    This is a heading with size "small" with a bottom margin
  </div>
</div>
`;

exports[`renders a Heading with size xlarge and a bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-xlarge);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
  margin-bottom: var(--echoes-dimension-space-250);
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    This is a heading with size "xlarge" with a bottom margin
  </div>
</div>
`;

exports[`renders a Heading with size xsmall and a bottom margin 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-xsmall);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
  margin-bottom: var(--echoes-dimension-space-100);
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    This is a heading with size "xsmall" with a bottom margin
  </div>
</div>
`;

exports[`uses the default size 1`] = `
.emotion-0 {
  font: var(--echoes-typography-heading-xlarge);
  color: var(--echoes-color-text-bold);
  margin: 0;
  max-width: var(--echoes-sizes-typography-max-width-default);
}

<div>
  <h1
    class="emotion-0 emotion-1"
  >
    Text goes here
  </h1>
</div>
`;
